### ISSUE TITLE
Global attribute setting bug

### ISSUE THREAD

**Issue by elyisgreat** [alda-lang/alda#256]
I'm not sure why, but in this score of mine, it is seeming to ignore the global attribute set

```
(tempo! 120)

piano "steven": (vol 90) o5
piano "background": (vol 60) o3
celesta: (vol 75) o6
clarinet: (vol 75) o5

#steven: [r1]*3 [r8]*5 (quant 30) f+8 f+ f+ f+4 e8~ d~ c+~ < b4~ > [r8]*6 (quant 90) f+8 f+ f+ f+4 e8 d c+ < b4 g8 g g > d d r d d d
#steven: (quant 30) c+8 < b16 > c+8 d c+ < b4. > [r16]*11 (quant 90) d8 d d (quant 30) c+ < b > c+16 d16. r32 c+16~8 < b8 b b (quant 90) g8. > d d8 r2
#background: [f+2/b/>d< e4./a/>c+< d8~1/g/b]*4 [f+2/b/>d< e4/a/>c+< e8/a/>c+< d8~1/g/b]*2
#celesta: [r1]*12
#clarinet: [r1]*12

#steven: r4 d8 c+4 c+8 d d < b1 > r4 d8 c+4 c+8 d f+8~2 e4 < a8 a8~4 > d8 d c+ c+4 d8 < b4. f+8 b a f+4~2.~8 a8 > d4 f+ c+ e d < b- > {e7 f+1} e8 d8~1~8 << d8 a > c+ e2 >
#background: e4./a/>c+< e8~2/a/>c+< d4./g/b d8~2/g/b e4./a/>c+< e8~2/a/>c+< f+2/a/>d< e/a/>c+< e4./a/>c+< e8~2/a/>c+< d4./g/b d8~2/g/b d1/f+/a
#background: f+2/b/>d< e/a/>c+< f+/b-/>d< e-/g/b-/>d< d1~1/f+/a/>c+<
#celesta: a4 r2.~4 c+8 d4 c+8 r4 a r2. [r1]*8
#clarinet: [r1]*3 f+2 g a1 r >f+< [r1]*4

steven: [r1]*8
background: [<b2>/f+/b/>d< <a4.>/e/a/>c+< <e8~1>/e/g/b]*4
celesta: [d2 c+4. <b8~1>]*2 r1~2 < d4 c+ <b>> r2.~1
clarinet: [r1]*3 r2~4~8~16 b16 >f+1~1< r4 b8 > f+2~8~4 b2 < r4 [r1]*3

steven: r4 c+8 c+4 c+8 d c+ < b2. > r2 d8 c+4 c+8 d4 f+2 e r4 c+8 c+4 d d8~4 < f+8 f+ b a4 f+2 r4 < f+8 a > c+ > e4 d c+ < b >
background: e4./a/>c+< e8~2/a/>c+< d4./g/b d8~2/g/b e4./a/>c+< e8~2/a/>c+< f+2/a/>d< e/a/>c+< e4./a/>c+< e8~2/a/>c+< d4./g/b d8~2/g/b d1~1/f+/a
celesta: a4 r2.~4 c+8 d4 c+8 r4 a r2. [r1]*5
clarinet: [r1]*8

steven: r4 d8 c+4 c+8 d d < b1 > r4 d8 c+4 c+8 d f+8~2 e4 < a8 a8~4 > d8 d c+ c+4 d8 < b4. f+8 b a f+4~2.~8 a8 > d4 f+ c+ e d < b- > f+ e8 d8~1 r2~4. < a8
background: e4./a/>c+< e8~2/a/>c+< d4./g/b d8~2/g/b e4./a/>c+< e8~2/a/>c+< f+2/a/>d< e/a/>c+< e4./a/>c+< e8~2/a/>c+< d4./g/b d8~2/g/b d1/f+/a
background: f+2/b/>d< e/a/>c+< f+/b-/>d< e-/g/b-/>d< d1~2./f+/a/>c+< r4
celesta: [r1]*11
clarinet: [r1]*11

(tempo! 110)

steven:  > d4 f+ c+ e d < b- > f2~2 e4 d4~1~1
background: f+2/b/>d< e/a/>c+< f+/b-/>d< g2~2/a/>c+/f< r2~8 f+2..~1/a/>c+/e<
celesta: [r1]*3 r8 e2..~1
clarinet: [r1]*5
```

When I comment out the block below, it correctly sets the global attribute

```
(tempo! 120)

piano "steven": (vol 90) o5
piano "background": (vol 60) o3
celesta: (vol 75) o6
clarinet: (vol 75) o5

#steven: [r1]*3 [r8]*5 (quant 30) f+8 f+ f+ f+4 e8~ d~ c+~ < b4~ > [r8]*6 (quant 90) f+8 f+ f+ f+4 e8 d c+ < b4 g8 g g > d d r d d d
#steven: (quant 30) c+8 < b16 > c+8 d c+ < b4. > [r16]*11 (quant 90) d8 d d (quant 30) c+ < b > c+16 d16. r32 c+16~8 < b8 b b (quant 90) g8. > d d8 r2
#background: [f+2/b/>d< e4./a/>c+< d8~1/g/b]*4 [f+2/b/>d< e4/a/>c+< e8/a/>c+< d8~1/g/b]*2
#celesta: [r1]*12
#clarinet: [r1]*12

#steven: r4 d8 c+4 c+8 d d < b1 > r4 d8 c+4 c+8 d f+8~2 e4 < a8 a8~4 > d8 d c+ c+4 d8 < b4. f+8 b a f+4~2.~8 a8 > d4 f+ c+ e d < b- > {e7 f+1} e8 d8~1~8 << d8 a > c+ e2 >
#background: e4./a/>c+< e8~2/a/>c+< d4./g/b d8~2/g/b e4./a/>c+< e8~2/a/>c+< f+2/a/>d< e/a/>c+< e4./a/>c+< e8~2/a/>c+< d4./g/b d8~2/g/b d1/f+/a
#background: f+2/b/>d< e/a/>c+< f+/b-/>d< e-/g/b-/>d< d1~1/f+/a/>c+<
#celesta: a4 r2.~4 c+8 d4 c+8 r4 a r2. [r1]*8
#clarinet: [r1]*3 f+2 g a1 r >f+< [r1]*4

#steven: [r1]*8
#background: [<b2>/f+/b/>d< <a4.>/e/a/>c+< <e8~1>/e/g/b]*4
#celesta: [d2 c+4. <b8~1>]*2 r1~2 < d4 c+ <b>> r2.~1
#clarinet: [r1]*3 r2~4~8~16 b16 >f+1~1< r4 b8 > f+2~8~4 b2 < r4 [r1]*3

steven: r4 c+8 c+4 c+8 d c+ < b2. > r2 d8 c+4 c+8 d4 f+2 e r4 c+8 c+4 d d8~4 < f+8 f+ b a4 f+2 r4 < f+8 a > c+ > e4 d c+ < b >
background: e4./a/>c+< e8~2/a/>c+< d4./g/b d8~2/g/b e4./a/>c+< e8~2/a/>c+< f+2/a/>d< e/a/>c+< e4./a/>c+< e8~2/a/>c+< d4./g/b d8~2/g/b d1~1/f+/a
celesta: a4 r2.~4 c+8 d4 c+8 r4 a r2. [r1]*5
clarinet: [r1]*8

steven: r4 d8 c+4 c+8 d d < b1 > r4 d8 c+4 c+8 d f+8~2 e4 < a8 a8~4 > d8 d c+ c+4 d8 < b4. f+8 b a f+4~2.~8 a8 > d4 f+ c+ e d < b- > f+ e8 d8~1 r2~4. < a8
background: e4./a/>c+< e8~2/a/>c+< d4./g/b d8~2/g/b e4./a/>c+< e8~2/a/>c+< f+2/a/>d< e/a/>c+< e4./a/>c+< e8~2/a/>c+< d4./g/b d8~2/g/b d1/f+/a
background: f+2/b/>d< e/a/>c+< f+/b-/>d< e-/g/b-/>d< d1~2./f+/a/>c+< r4
celesta: [r1]*11
clarinet: [r1]*11

(tempo! 110)

steven:  > d4 f+ c+ e d < b- > f2~2 e4 d4~1~1
background: f+2/b/>d< e/a/>c+< f+/b-/>d< g2~2/a/>c+/f< r2~8 f+2..~1/a/>c+/e<
celesta: [r1]*3 r8 e2..~1
clarinet: [r1]*5
```

I haven't been able to find the base case yet, but if I can find it I will put it as a comment in this issue.

What do you think is going on?

**Comment by daveyarwood** [comment]
So, my first thought was that it's difficult to hear the difference between 120 and 110, so I changed the 110 to 60 so the change would be more noticeable, but sure enough, it seemed to stay at 120 the whole time.

Then I tried commenting out the block you commented out, and I _did_ hear the change to 60 bpm. So I started looking further into that block.

One thing that it helps to understand is that when you use global attributes "between" instrument parts like this, really they are still part of a particular instrument part, and it's the last one you used. So in this case, the clarinet part owns the tempo change. In other words, the tempo change will happen wherever the clarinet's next note/rest would happen. So you aren't hearing the main melody (the "steven" part) slow down because that's not where the clarinet part is.

I think the mismatch is in this line:

```
clarinet: [r1]*3 r2~4~8~16 b16 >f+1~1< r4 b8 > f+2~8~4 b2 < r4 [r1]*3
```

Judging by the "steven" part in this block, each part should last 8 measures. However, this line lasts 11 measures, so it skews the clarinet part in each block forward by 3 measures.

Then, when you get down to this part:

```
clarinet: [r1]*11

(tempo! 110)
```

The clarinet part sets the global tempo to 110, 3 bars further ahead than everyone else is at that point in the score. And the last block is about 3 bars long, so I think the tempo change is still happening, but we don't hear it because it's happening right at the end of the score.

I removed `[r1]*3` from the end of that one clarinet line above, and tried playing it again, and I heard the tempo change on that last line.
